flake.nix

```diff
@@ -0,0 +1,40 @@
+{
+  description = "Rust flake template";
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    utils.url = "github:numtide/flake-utils";
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { self, nixpkgs, utils, fenix }:
+    utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = [
+            fenix.overlay
+          ];
+        };
+      in
+      {
+
+
+        devShell = with pkgs; mkShell {
+          shellHook = ''
+          '';
+          buildInputs = with pkgs; [
+            (pkgs.fenix.complete.withComponents [
+              "cargo"
+              "clippy"
+              "rust-src"
+              "rustc"
+              "rustfmt"
+              "rust-analyzer"
+            ])
+          ];
+        };
+      });
+}
```

Comment:
Please remove this file again.

Reply:
Arg. I thought I didn't commit that file. Sorry about that.

Reply:
Do you also want me to squash these commits down into one, so that the commit history looks less messy?

Reply:
Not necessarily, PRs are squashed on merge.

Reply:
Alright, perfect.